### PR TITLE
SSZ wiki page 

### PR DESCRIPTION
**Pull request by thogiti**
## Wiki PR Checklist

- [x] SSZ Wiki page
- [ ] Merkleization wiki page

### CODE REVIEW

**Review by taxmeifyoucan — APPROVED** [review]
Thanks for the PR! It looks almost done? Just 2 things I would elaborate on more: 
CL uses merkle roots for everything instead of just hashes and enables light clients to use easy proofs. The document should also explain merklization, maybe pointing to Proto's resources https://github.com/protolambda/eth2-docs#ssz-hash-tree-root-and-merkleization
And one more section I would add is a comparison with RLP - what are the main benefits, features over RLP and hint that there is an effort to move everything to ssz.

**Comment by thogiti** [comment]
> Thanks for the PR! It looks almost done? Just 2 things I would elaborate on more: CL uses merkle roots for everything instead of just hashes and enables light clients to use easy proofs. The document should also explain merklization, maybe pointing to Proto's resources https://github.com/protolambda/eth2-docs#ssz-hash-tree-root-and-merkleization And one more section I would add is a comparison with RLP - what are the main benefits, features over RLP and hint that there is an effort to move everything to ssz.

Thank you @taxmeifyoucan I added a section for SSZ VS RLP comparison and briefly explained the benefits of SSZ over RLP and the intent to completely adopt SSZ.

I am now working on the SSZ merkleization. This will complete this wiki page.

**Comment by thogiti** [comment]
I am separating the merkleization content into a separate page. The SSZ page is already large.

File: docs/wiki/CL/SSZ.md [hunk 1]
```diff
@@ -0,0 +1,825 @@
...
+
+3. **Indexing**:
+   - **SSZ**: Although indexing is described as 'poor', SSZ supports some level of direct access to serialized data without full deserialization, which is beneficial for certain operations within the blockchain.
+   - **RLP**: Does not support efficient indexing, potentially leading to `O(N)` complexity when accessing internal data, which can be a significant drawback for performance on large-scale networks.
+
+4. **Data Type Compatibility**:
+   - **SSZ**: Designed to be fully compatible with the data types and structures used within the Ethereum protocol, enhancing its utility for consensus mechanisms and network operations.
+   - **RLP**: While flexible, the need for additional layers to support various data types can lead to inefficiencies and increased complexity in implementation.
+
+5. **Designed for Ethereum 2.0 Requirements**:
```
[the remark this answers is not in the document]
Comment (thogiti) [reply docs/wiki/CL/SSZ.md]: This commit will address the above two changes. https://github.com/eth-protocol-fellows/protocol-studies/pull/237/commits/db0c2af347256f5fe5609b7d8fd82d5f81ea9711

**Pull request merged**